src/utils/routes.ts

```diff
@@ -52,11 +49,22 @@ export const getRoutesFromPath = (path: unknown, functionName: string, methods:
     // for both `/foo` and `/foo/`.
     const normalizedRegex = `^${regex}\\/?$`
 
-    return [{ pattern: path, expression: normalizedRegex, methods }]
+    return { pattern: path, expression: normalizedRegex, methods }
   } catch {
     throw new FunctionBundlingUserError(`'${path}' is not a valid path according to the URLPattern specification`, {
       functionName,
       runtime: RUNTIME.JAVASCRIPT,
     })
   }
 }
+
+export const getRoutes = (input: unknown, functionName: string, methods: string[]): Route[] => {
+  if (!input) {
+    return []
+  }
+
+  const paths = Array.isArray(input) ? input : [input]
+  const routes = paths.map((path) => getRoute(path, functionName, methods ?? [])).filter(nonNullable)
+
+  return routes
```

Comment:
I would do `return [...new Set(routes)]` to remove duplicates

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/1586/commits/9e2cac92e2a50eb82f1ae5f112651b3933522d81.

Reply:
We can't do it on `routes` because it's an array of objects. I've done it in `paths`.